src/components/navigation/navigation-item/navigation-item.component.ts

```diff
@@ -0,0 +1,138 @@
+import {
+    Component,
+    Input,
+    ViewEncapsulation,
+    ContentChildren,
+    QueryList,
+    Renderer2,
+    ElementRef,
+    Optional,
+    SkipSelf,
+    ContentChild,
+    Output,
+    AfterViewInit,
+  HostListener
+} from '@angular/core';
+import { Router, RouterLinkActive, NavigationEnd, RouterState, ActivatedRoute, UrlTree } from '@angular/router';
+import 'rxjs/add/operator/filter';
```

Comment:
Instead of using this rxjs import here, it is not recommended to use lettable operators instead. [Read this for an intro](https://blog.hackages.io/rxjs-5-5-piping-all-the-things-9d469d1b3f44).

This would change the import to something like:

```typescript
import { filter } from 'rxjs/operators/filter';

...

_router.events.pipe(filter(event => event instanceof NavigationEnd)).subscribe...
```

Reply:
Lettable operator used as suggested.